itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/TestMetastoreHousekeepingLeader.java

```diff
@@ -34,7 +35,7 @@ public class TestMetastoreHousekeepingLeader extends MetastoreHousekeepingLeader
 
   @Before
   public void setUp() throws Exception {
-    internalSetup("localhost", true);
+    internalSetup("localhost", MetastoreConf.newMetastoreConf());
```

Comment:
do you need extra conf param? it seems every tests passes the same thing

Reply:
It's used here: https://github.com/apache/hive/pull/6142/files#diff-bd73307ca500a5ceb6b862f1819eb919a4f03f8db1737fad425a1b684353c713R52, mainly for sharing the same connection url for different leader voter

Reply:
Interesting approach to instantiate and use a junit TEST class within another test. 
https://github.com/apache/hive/pull/6142/files#diff-bd73307ca500a5ceb6b862f1819eb919a4f03f8db1737fad425a1b684353c713R46 
It might be cleaner to extract the service logic into a separate class.

Reply:
Move the check to the parent class, https://github.com/apache/hive/pull/6142/files#diff-afdd9d2455383922b9eb4eaa0bae23dc39cd01a081100d00acb16b2e41322635R234